standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -546,7 +552,11 @@ public static void updateTableStatsForCreateTable(Warehouse wh, Database db, Tab
       }
     }
 
-    if (MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.STATS_AUTO_GATHER)) {
+    String statsSource = Optional.ofNullable(envContext).orElse(new EnvironmentContext(Collections.emptyMap()))
+            .getProperties().getOrDefault(HIVE_ICEBERG_STATS_SOURCE, HiveMetaHook.ICEBERG.toLowerCase()).toLowerCase();
+    if (MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.STATS_AUTO_GATHER) && 
+        !(isIcebergTable && statsSource.equals(HiveMetaHook.ICEBERG.toLowerCase()))) {
```

Comment:
why we do lowercase everywhere, i think we should introduce enum for the statsSource

Reply:
introduced enum for the statsSource